ThermodynamicsDry.pyx

```diff
@@ -141,20 +144,35 @@ cdef class ThermodynamicsDry:
             Py_ssize_t jmax = Gr.dims.nlg[1] - Gr.dims.gw
             Py_ssize_t kmax = Gr.dims.nlg[2] - Gr.dims.gw
             Py_ssize_t count
-            Py_ssize_t s_shift = PV.get_varshift(Gr,'s')
+            Py_ssize_t s_shift
             double [:] data = np.empty((Gr.dims.npl,),dtype=np.double,order='c')
 
         #Add entropy potential temperature to 3d fields
-        with nogil:
-            count = 0
-            for i in xrange(imin,imax):
-                ishift = i * istride
-                for j in xrange(jmin,jmax):
-                    jshift = j * jstride
-                    for k in xrange(kmin,kmax):
-                        ijk = ishift + jshift + k
-                        data[count] = thetas_c(PV.values[s_shift+ijk],0.0)
-                        count += 1
+        if self.s_prognostic:
+            s_shift = PV.get_varshift(Gr,'s')
+            with nogil:
+                count = 0
+                for i in xrange(imin,imax):
+                    ishift = i * istride
+                    for j in xrange(jmin,jmax):
+                        jshift = j * jstride
+                        for k in xrange(kmin,kmax):
+                            ijk = ishift + jshift + k
+                            data[count] = thetas_c(PV.values[s_shift+ijk],0.0)
+                            count += 1
+
+        else:
+            s_shift = DV.get_varshift(Gr,'s')
+            with nogil:
+                count = 0
+                for i in xrange(imin,imax):
+                    ishift = i * istride
+                    for j in xrange(jmin,jmax):
+                        jshift = j * jstride
+                        for k in xrange(kmin,kmax):
+                            ijk = ishift + jshift + k
+                            data[count] = thetas_c(DV.values[s_shift+ijk],0.0)
+                            count += 1
```

Comment:
I guess this is supposed to be related to thetali but it looks exactly the same as the code below if self.s_prognostic.